envoy.dependency.check/envoy/dependency/check/abstract/checker.py

```diff
@@ -332,6 +356,18 @@ async def preload_release_dates(self) -> None:
         async for dep in preloader:
             self.log.debug(f"Preloaded release date: {dep.id}")
 
+    @checker.preload(
+        when=["sha"],
+        unless=["releases", "release_issues", "release_dates"],
+        catches=[ConcurrentError, gidgethub.GitHubException])
+    async def preload_release_shas(self) -> None:
+        preloader = inflate(
+            self.github_dependencies,
+            lambda d: (
+                d.release.sha, ))
```

Comment:
i think we need the `limit` here i doubt anything more than 4 will improve performance, and may well be why im seeing some connect issues locally